src/cctag/filter/cvRecode.cpp

```diff
@@ -241,113 +231,24 @@ void cvRecodedCanny(
-      CV_MAT_ELEM( *kerneldX, float, 8, 4 ) = 0.f  ;
-      CV_MAT_ELEM( *kerneldX, float, 8, 5 ) = -0.000064765993382f  ;
-      CV_MAT_ELEM( *kerneldX, float, 8, 6 ) = -0.000028902492951f  ;
-      CV_MAT_ELEM( *kerneldX, float, 8, 7 ) = -0.000003558691641f  ;
-      CV_MAT_ELEM( *kerneldX, float, 8, 8 ) = -0.000000143284235f  ;
-
-      cvConvertScale( kerneldX, kerneldX, -1.f );
-      cvTranspose( kerneldX, kerneldY );
-
-      cvFilter2D( src, dx, kerneldX );
-      cvFilter2D( src, dy, kerneldY );
-
-//      CCTAG_COUT("DX_DEBUG values");
-//      CCTAG_COUT(dx_debug->rows);
-//      for (int i=0; i< dx->rows ; ++i)
-//      {
-//        for (int j=0; j< dx->cols ; ++j)
-//        {
-//          std::cout << dx->data.s[ i*dx->step + j] << " ";
-//        }
-//        std::cout << std::endl;
-//      }
-//      CCTAG_COUT("END DX_DEBUG values");
-      
-      cvReleaseMat( &kerneldX );
-      cvReleaseMat( &kerneldY );
+      // The second option is to apply the (9x9) 2D following kernel
+      cv::Mat kerneldX = (cv::Mat_<float>(9,9) << 
+        0.000000143284235f, 0.000003558691641f, 0.000028902492951f, 0.000064765993382f, 0.f, -0.000064765993382f, -0.000028902492951f, -0.000003558691641f, -0.000000143284235f,
+        0.000004744922188f, 0.000117847682078f, 0.000957119116802f, 0.002144755142391f, 0.f, -0.002144755142391f, -0.000957119116802f, -0.000117847682078f, -0.000004744922188f,
+        0.000057804985902f, 0.001435678675203f, 0.011660097860113f, 0.026128466569370f, 0.f, -0.026128466569370f, -0.011660097860113f, -0.001435678675203f, -0.000057804985902f,
+        0.000259063973527f, 0.006434265427174f, 0.052256933138740f, 0.117099663048638f, 0.f, -0.117099663048638f, -0.052256933138740f, -0.006434265427174f, -0.000259063973527f,
+        0.000427124283626f, 0.010608310271112f, 0.086157117207395f, 0.193064705260108f, 0.f, -0.193064705260108f, -0.086157117207395f, -0.010608310271112f, -0.000427124283626f,
+        0.000259063973527f, 0.006434265427174f, 0.052256933138740f, 0.117099663048638f, 0.f, -0.117099663048638f, -0.052256933138740f, -0.006434265427174f, -0.000259063973527f,
+        0.000057804985902f, 0.001435678675203f, 0.011660097860113f, 0.026128466569370f, 0.f, -0.026128466569370f, -0.011660097860113f, -0.001435678675203f, -0.000057804985902f,
+        0.000004744922188f, 0.000117847682078f, 0.000957119116802f, 0.002144755142391f, 0.f, -0.002144755142391f, -0.000957119116802f, -0.000117847682078f, -0.000004744922188f,
+        0.000000143284235f, 0.000003558691641f, 0.000028902492951f, 0.000064765993382f, 0.f, -0.000064765993382f, -0.000028902492951f, -0.000003558691641f, -0.000000143284235f);
+
+      kerneldX.convertTo(kerneldX, CV_32FC1, -1.f);
+      cv::Mat kerneldY;
+      cv::transpose( kerneldX, kerneldY );
+
+      cv::filter2D(src, dx, CV_16SC1, kerneldX, cv::Point(-1, -1), 0.0, cv::BORDER_REPLICATE);
+      cv::filter2D(src, dy, CV_16SC1, kerneldY, cv::Point(-1, -1), 0.0, cv::BORDER_REPLICATE);    
```

Comment:
```suggestion
      const cv::Point anchor{-1,-1};
      const double delta{0};
      cv::filter2D(src, dx, CV_16SC1, kerneldX, anchor, delta, cv::BORDER_REPLICATE);
      cv::filter2D(src, dy, CV_16SC1, kerneldY, anchor, delta, cv::BORDER_REPLICATE);
```
it's more a style/readability thing, so you always know what are you passing